lib/authCodeGrant.js

```diff
@@ -172,7 +173,11 @@ function generateCode (done) {
  */
 function saveAuthCode (done) {
   var expires = new Date();
-  expires.setSeconds(expires.getSeconds() + this.config.authCodeLifetime);
+
+  var ttl = !isNaN(parseFloat(this.client.authCodeLifetime)) && isFinite(this.client.authCodeLifetime) ?
+    this.client.authCodeLifetime : this.config.authCodeLifetime;
+
+  expires.setSeconds(expires.getSeconds() + ttl);
```

Comment:
I think we should extract this into a utility function, maybe:

```js
Grant.prototype.setExpires = function (expires, key) {
  var expires = new Date();

  var ttl = !isNaN(parseFloat(this.client[key])) && isFinite(this.client[key]) ? this.client[key]: this.config[key];

  expires.setSeconds(expires.getSeconds() + ttl);

  return expires;
};

```

What do you think?